### PR TITLE
make buttons work

### LINKED ISSUE #90: make the continue button work
connect the details page to login page 
connect back pages

### LINKED ISSUE #95: apear the splash page if login if not got to upload photo

### PR DESCRIPTION
**Pull request by amanshawar**
the back button back to previous page
the green button  in get details go to the /item-list if logged if not go to /login-form
the first page not apear if is logged in
fix #95 
fix #90

### CODE REVIEW

File: client/src/components/ItemList/index.js [hunk 1]
```diff
@@ -4,25 +4,39 @@ import Title from '../Shared/Title';
 import GButton from '../Shared/GreenButton';
 import Footer from '../Shared/Footer';
 import Item from '../Item';
-import { List, StyledHeader, StyledBottom, StyledLink, GButtonContainer } from './itemlist.style';
+import {
+  List, StyledHeader, StyledBottom, StyledLink, GButtonContainer,
+} from './itemlist.style';
+
 class ItemList extends Component {
+  constructor(props) {
```
Comment (jema28) [inline client/src/components/ItemList/index.js]: This constructor isn't needed.

File: client/src/components/Shared/Button/index.js [hunk 2]
```diff
@@ -1,5 +1,22 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
 import StyledButton from './button.style';
 
-const Button = () => <StyledButton type="button">BACK</StyledButton>;
-export default Button;
+
+class Button extends Component {
+  constructor(props) {
```
Comment (jema28) [inline client/src/components/Shared/Button/index.js]: This doesn't have to be a class component. Have a look at the component Hadeel wrote today.

**Review by jema28 — CHANGES_REQUESTED** [review]
Here are the first comments I have. Will carry on reviewing now.

**Review by jema28 — APPROVED** [review]

File: client/src/components/ItemList/index.js [hunk 3]
```diff
@@ -4,25 +4,33 @@ import Title from '../Shared/Title';
...
 
   componentDidMount() {
     axios.get('/items')
-      .then(res => this.setState({itemlist: res.data.data}))
-      .catch(err => console.log(err))
+      .then(res => this.setState({ itemlist: res.data.data }))
+      .catch(err => console.log(err));
   }
 
+  uploadPhoto = () => {
```
Comment (Hadeel-Salamin) [inline client/src/components/ItemList/index.js]: the name of the function is tricky :p 
it would be better if its called goToUploadPage or just addNewItem

Comment (jema28) [reply client/src/components/ItemList/index.js]: Good point.

**Review by jema28 — APPROVED** [review]

**Pull request merged**